scheduler-driver/src/main/java/gov/nasa/jpl/aerie/scheduler/solver/PrioritySolver.java

```diff
@@ -812,11 +858,33 @@ private Collection<SchedulingActivityDirective> getBestNewActivities(MissingActi
         //REVIEW: not yet handling multiple activities at a time
         logger.info("Instantiating activity in windows " + startWindows.trueSegmentsToString());
         final var act = createOneActivity(
-            missingTemplate.getActTemplate(),
+            missingTemplate,
             goal.getName() + "_" + java.util.UUID.randomUUID(),
             startWindows,
             missing.getEvaluationEnvironment());
-        act.ifPresent(newActs::add);
+        if(act.isPresent()){
+          if (missingTemplate.getAnchorId().isPresent()) {
+            SchedulingActivityDirective predecessor = plan.getActivitiesById().get(missingTemplate.getAnchorId().get());
+            int includePredDuration = 0;
+            if (missingTemplate.getAnchorToStart().isPresent()) {
+              includePredDuration = missingTemplate.getAnchorToStart().get() ? 0 : 1;
+            }
+            Duration dur = predecessor.duration().times(includePredDuration);
+            Duration startOffset = act.get().startOffset().minus(plan.calculateAbsoluteStartOffsetAnchoredActivity(predecessor).plus(dur));
```

Comment:
`predecessor`, `dur`, `startOffset` could be `final var`

Reply:
fixed